### PR TITLE
simplify

### PR DESCRIPTION
**Pull request by dongjoon-hyun**

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala [hunk 1]
```diff
@@ -117,29 +115,16 @@ object OrcUtils extends Logging {
...
-                }
-            }.getOrElse(-1)
-          })
+        val resolver = if (isCaseSensitive) caseSensitiveResolution else caseInsensitiveResolution
+        // Need to fail if there is ambiguity, i.e. more than one field is matched.
+        requiredSchema.fieldNames.foreach { requiredFieldName =>
+          val matchedOrcFields = orcFieldNames.filter(resolver(_, requiredFieldName))
+          if (matchedOrcFields.size > 1) {
+            throw new RuntimeException(s"""Found duplicate field(s) "$requiredFieldName": """ +
+              s"${matchedOrcFields.mkString("[", ", ", "]")} in case-insensitive mode")
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcUtils.scala:124]: Actually, we do this in both modes, but `orcFieldNames` doesn't have case-sensitively duplicated names.

**Comment by dongjoon-hyun** [comment]
@seancxmao . I tested this only on your new test case.

**Comment by seancxmao** [comment]
@dongjoon-hyun Thanks! I merged your PR. I tested with more test cases and all tests passed.

```
build/sbt "sql/test-only *.FileBasedDataSourceSuite *.OrcQuerySuite *.SQLQuerySuite"
build/sbt "hive/test-only *.HiveOrcQuerySuite *.HiveQuerySuite"
```

**Comment by dongjoon-hyun** [comment]
Thanks!

**Pull request merged**